subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -204,32 +226,33 @@ public void setDependencySpec(@Nullable Spec<DependencyResult> dependencySpec) {
      * and groovy closures. Spec and closure receive {@link DependencyResult} as parameter.
      * Examples of String notation: 'org.slf4j:slf4j-api', 'slf4j-api', or simply: 'slf4j'.
      * The input may potentially match multiple dependencies.
-     * See also {@link #setDependencySpec(Spec)}
      * <p>
      * This method is exposed to the command line interface. Example usage:
      * <pre>gradle dependencyInsight --dependency slf4j</pre>
+     *
+     * @since 9.0
      */
+    @Input
+    @Optional
     @Option(option = "dependency", description = "Shows the details of given dependency.")
-    public void setDependencySpec(@Nullable Object dependencyInsightNotation) {
-        NotationParser<Object, Spec<DependencyResult>> parser = DependencyResultSpecNotationConverter.parser();
-        setDependencySpec(parser.parseNotation(dependencyInsightNotation));
-    }
+    @ReplacesEagerProperty(replacedAccessors = {
+        @ReplacedAccessor(value = ReplacedAccessor.AccessorType.SETTER, name = "setDependencySpec", originalType = Object.class)
+    })
+    public abstract Property<Object> getDependencyNotation();
 
     /**
      * Configuration to look the dependency in
      */
     @Internal
-    @ToBeReplacedByLazyProperty
-    public @Nullable Configuration getConfiguration() {
-        return configuration;
+    @ReplacesEagerProperty
+    public Property<Configuration> getConfiguration() {
+        return Objects.requireNonNull(configurationProp.get());
     }
 
-    /**
-     * Sets the configuration to look the dependency in.
-     */
-    public void setConfiguration(@Nullable Configuration configuration) {
-        this.configuration = configuration;
-    }
+    // used by CC to have serializable results
+    @SuppressWarnings("ClassEscapesDefinedScope")
+    @Internal
+    protected abstract Property<SerializableConfigurationData> getConfigurationDetails();
```

Comment:
❓ Doe we need to expose that as a protected method, or can we have a private field?